statemachine.go

```diff
@@ -964,18 +978,21 @@ func runStateMachineForServiceUser(
 func runStateMachineForServiceProvider(
 	conn net.Conn,
 	upcallCh chan upcallEvent,
+	upcallStreamCh chan upcallEvent,
 	downcallCh chan stateEvent,
 	label string) {
 	sm := &stateMachine{
-		label:          label,
-		isUser:         false,
-		contextManager: newContextManager(label),
-		conn:           conn,
-		netCh:          make(chan stateEvent, 128),
-		errorCh:        make(chan stateEvent, 128),
-		downcallCh:     downcallCh,
-		upcallCh:       upcallCh,
-		faults:         getProviderFaultInjector(),
+		label:            label,
+		isUser:           false,
+		contextManager:   newContextManager(label),
+		conn:             conn,
+		netCh:            make(chan stateEvent, 128),
+		errorCh:          make(chan stateEvent, 128),
+		downcallCh:       downcallCh,
+		upcallCh:         upcallCh,
+		upcallStreamCh:   upcallStreamCh,
+		faults:           getProviderFaultInjector(),
+		commandAssembler: dimse.NewCommandAssembler(),
```

Comment:
We need to add the same line to `runStateMachineForServiceUser`? Because if not, the `commandAssembler` will be init with default values and using the `stream` channel will cause panic.